test/internal/k8sutils/utils.go

```diff
@@ -267,6 +267,47 @@ func WaitForPodDeployment(ctx context.Context, clientset *kubernetes.Clientset,
 	return errors.Wrapf(retrier.Do(ctx, checkPodDeploymentFn), "could not wait for deployment %s", deploymentName)
 }
 
+func WaitForPodDaemonset(ctx context.Context, clientset *kubernetes.Clientset, namespace, daemonsetName, podLabelSelector string) error {
+	podsClient := clientset.CoreV1().Pods(namespace)
+	daemonsetClient := clientset.AppsV1().DaemonSets(namespace)
+	checkPodDaemonsetFn := func() error {
+		daemonset, err := daemonsetClient.Get(ctx, daemonsetName, metav1.GetOptions{})
+		if err != nil {
+			return errors.Wrapf(err, "could not get daemonset %s", daemonsetName)
+		}
+
+		if daemonset.Status.NumberReady == 0 && daemonset.Status.DesiredNumberScheduled == 0 {
+			// Capture daemonset restart. Restart sets every numerical status to 0.
+			log.Printf("daemonset %s has %d pods in ready status, expected %d", daemonsetName, daemonset.Status.NumberReady, daemonset.Status.DesiredNumberScheduled)
```

Comment:
Won't this always print 0 for ready and expected ?
better to log something like daemonset is in restart phase or something.